src/Client/LiveStripeClient.php

```diff
@@ -44,8 +42,21 @@ public function createPaymentIntent(array $createPayload): PaymentIntent
         return $this->stripeClient->paymentIntents->create($createPayload);
     }
 
-    public function retrievePaymentMethod(string $paymentMethodId): PaymentMethod
+    public function updatePaymentMethodBillingDetail(string $paymentMethodId, Donation $donation): PaymentMethod
     {
-        return $this->stripeClient->paymentMethods->retrieve($paymentMethodId);
+        // "A PaymentMethod must be attached a customer to be updated."
+        $this->stripeClient->paymentMethods->attach($paymentMethodId, ['customer' => $donation->getPspCustomerId()]);
+
+        // Address etc. is set up in Stripe.js already. Adding these values which we collect on the
+        // donation separately helps with support queries and maybe with fraud signals.
+        return $this->stripeClient->paymentMethods->update(
+            $paymentMethodId,
+            [
+                'billing_details' => [
+                    'name' => "{$donation->getDonorFirstName()} {$donation->getDonorLastName()}",
```

Comment:
I think it's good if we can keep this LiveStripeClient class as stupid as possible, so it doesn't need testing, since it's coupled to the Stripe API and a bit awkward to test. So I'd suggest even concatenating the name parts is too much to do here. Can we possibly change the function signature and do one of these:

```suggestion
                    'name' => $donation->getDonorFullName()
```

or 

```suggestion
                    'name' => $fullName // param of the present function
```

or

```suggestion
                    'name' => $donorName->value, // new value object class
```

Or maybe

```
     return $this->stripeClient->paymentMethods->update(
            $paymentMethodId,
            [
                'billing_details' => [
                    'name' => $billingDetails->name,
                    'email' => $billingDetails->email,
                ],
            ],
        );

Reply:
Makes sense. I just went with a helper on the domain model for now to improve this reasonably quickly, as there are a few PRs in flight atm. https://github.com/thebiggive/matchbot/pull/726/commits/1f659d60a8536373f80029fc3a6875890e71d7d6